src/Settings/Default.hs-boot

```diff
@@ -3,8 +3,8 @@ module Settings.Default (
     defaultArgs, defaultLibraryWays, defaultRtsWays, defaultFlavour, defaultSplitObjects
     ) where
 
-import Flavour
-import Expression
+import Types.Flavour
+import Types.Expression
```

Comment:
Structuring the module namespace this way seems a bit odd; wouldn't it be more natural for these to be, e.g., `Flavour.Types`? Afterall, `Types.Flavour` has more in common with `Flavour` that, for instance, `Types.Expression`.

Reply:
I second that. We then can also get rid of plural, getting `Expression.Type`, `Flavour.Type`, etc.

Reply:
I also would like this to be factored out into a separate PR.

Reply:
sure, why not. I don't really mind where the modules live. It's just that data-types and alike are often leaf nodes, and if coupled too tightly with the logic around them quickly lead to cyclic imports.

Reply:
@alpmestan is working on a PR for this right now.